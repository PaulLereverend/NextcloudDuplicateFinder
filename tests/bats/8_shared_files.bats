#!/usr/bin/env bats
TESTSUITE="duplicatefinder"
load helper.sh

createShare() {
  curl -u "tuser:${OC_PASS}" -H "OCS-APIRequest: true" "${NEXTCLOUD}/ocs/v1.php/apps/files_sharing/api/v1/shares" $@
}

deleteShare() {
  shareId=$1
  shift
  curl -X DELETE -u "tuser:${OC_PASS}" -H "OCS-APIRequest: true" "${NEXTCLOUD}/ocs/v1.php/apps/files_sharing/api/v1/shares/${shareId}" $@
}

setup() {
  if [ -z "$NEXTCLOUD" ]; then
    echo "environment variable NEXTCLOUD need to bee defined"
    exit 1
  fi
  ./occ app:enable files_sharing

  export MAX_FILES=3
  load ${BATS_TEST_DIRNAME}/setup.sh
  createTestFiles $MAX_FILES \
    'data/tuser/files/userShareTest' \
    'data/tuser/files/userShareTest/subdir' \
    'data/tuser/files/userShareTest/subdir/subsubdir' \
    'data/tuser/files/groupShareTest' \
    'data/tuser/files/groupShareTest/subdir'\
    'data/tuser/files/groupShareTest/subdir/subsubdir'

  
  userShareId=$(createShare -d path="userShareTest" -d shareType=0 -d shareWith=admin | grep '<id>' | sed 's/\s*<id>\(.*\)<\/id>/\1/')
  groupShareId=$(createShare -d path="groupShareTest" -d shareType=1 -d shareWith=admin | grep '<id>' | sed 's/\s*<id>\(.*\)<\/id>/\1/')
  export userShareId groupShareId
}

teardown(){
    clearTestFiles 'data/tuser/files/groupShareTest' 'data/tuser/files/userShareTest'
    deleteShare $userShareId
    deleteShare $groupShareId
}

@test "[$TESTSUITE] Test for shared files" {
  output=$(./occ -v duplicates:find-all -u admin)
  expectedHash="07c5a378a726128905271919870b4d0422ed40e10bfb4df4708545a529d5885b"
  evaluateHashResult "${expectedHash}" 4 "${output}"
}
