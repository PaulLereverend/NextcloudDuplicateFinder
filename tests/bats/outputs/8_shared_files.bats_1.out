Start searching files for admin in path /admin/files
Scanning /admin/files/tests/3_3.txt
Scanning /admin/files/tests/0_0.txt
Scanning /admin/files/tests/3_2.txt
Scanning /admin/files/tests/1_1.txt
Scanning /admin/files/tests/2_0.txt
Scanning /admin/files/tests/1_0.txt
Scanning /admin/files/tests/3_1.txt
Scanning /admin/files/tests/2_2.txt
Scanning /admin/files/tests/3_0.txt
Scanning /admin/files/tests/2_1.txt
Scanning Shared Node /tuser/files/userShareTest/3_3.txt
Scanning Shared Node /tuser/files/userShareTest/0_0.txt
Scanning Shared Node /tuser/files/userShareTest/3_2.txt
Scanning Shared Node /tuser/files/userShareTest/1_1.txt
Scanning Shared Node /tuser/files/userShareTest/2_0.txt
Scanning Shared Node /tuser/files/userShareTest/1_0.txt
Scanning Shared Node /tuser/files/userShareTest/3_1.txt
Scanning Shared Node /tuser/files/userShareTest/2_2.txt
Scanning Shared Node /tuser/files/userShareTest/3_0.txt
Scanning Shared Node /tuser/files/userShareTest/2_1.txt
Scanning Shared Node /tuser/files/userShareTest/subdir/3_3.txt
Scanning Shared Node /tuser/files/userShareTest/subdir/0_0.txt
Scanning Shared Node /tuser/files/userShareTest/subdir/3_2.txt
Scanning Shared Node /tuser/files/userShareTest/subdir/1_1.txt
Scanning Shared Node /tuser/files/userShareTest/subdir/2_0.txt
Scanning Shared Node /tuser/files/userShareTest/subdir/1_0.txt
Scanning Shared Node /tuser/files/userShareTest/subdir/3_1.txt
Scanning Shared Node /tuser/files/userShareTest/subdir/2_2.txt
Scanning Shared Node /tuser/files/userShareTest/subdir/3_0.txt
Scanning Shared Node /tuser/files/userShareTest/subdir/2_1.txt
Scanning Shared Node /tuser/files/userShareTest/subdir/subsubdir/3_3.txt
Scanning Shared Node /tuser/files/userShareTest/subdir/subsubdir/0_0.txt
Scanning Shared Node /tuser/files/userShareTest/subdir/subsubdir/3_2.txt
Scanning Shared Node /tuser/files/userShareTest/subdir/subsubdir/1_1.txt
Scanning Shared Node /tuser/files/userShareTest/subdir/subsubdir/2_0.txt
Scanning Shared Node /tuser/files/userShareTest/subdir/subsubdir/1_0.txt
Scanning Shared Node /tuser/files/userShareTest/subdir/subsubdir/3_1.txt
Scanning Shared Node /tuser/files/userShareTest/subdir/subsubdir/2_2.txt
Scanning Shared Node /tuser/files/userShareTest/subdir/subsubdir/3_0.txt
Scanning Shared Node /tuser/files/userShareTest/subdir/subsubdir/2_1.txt
Scanning Shared Node /tuser/files/groupShareTest/3_3.txt
Scanning Shared Node /tuser/files/groupShareTest/0_0.txt
Scanning Shared Node /tuser/files/groupShareTest/3_2.txt
Scanning Shared Node /tuser/files/groupShareTest/1_1.txt
Scanning Shared Node /tuser/files/groupShareTest/2_0.txt
Scanning Shared Node /tuser/files/groupShareTest/1_0.txt
Scanning Shared Node /tuser/files/groupShareTest/3_1.txt
Scanning Shared Node /tuser/files/groupShareTest/2_2.txt
Scanning Shared Node /tuser/files/groupShareTest/3_0.txt
Scanning Shared Node /tuser/files/groupShareTest/2_1.txt
Scanning Shared Node /tuser/files/groupShareTest/subdir/3_3.txt
Scanning Shared Node /tuser/files/groupShareTest/subdir/0_0.txt
Scanning Shared Node /tuser/files/groupShareTest/subdir/3_2.txt
Scanning Shared Node /tuser/files/groupShareTest/subdir/1_1.txt
Scanning Shared Node /tuser/files/groupShareTest/subdir/2_0.txt
Scanning Shared Node /tuser/files/groupShareTest/subdir/1_0.txt
Scanning Shared Node /tuser/files/groupShareTest/subdir/3_1.txt
Scanning Shared Node /tuser/files/groupShareTest/subdir/2_2.txt
Scanning Shared Node /tuser/files/groupShareTest/subdir/3_0.txt
Scanning Shared Node /tuser/files/groupShareTest/subdir/2_1.txt
Scanning Shared Node /tuser/files/groupShareTest/subdir/subsubdir/3_3.txt
Scanning Shared Node /tuser/files/groupShareTest/subdir/subsubdir/0_0.txt
Scanning Shared Node /tuser/files/groupShareTest/subdir/subsubdir/3_2.txt
Scanning Shared Node /tuser/files/groupShareTest/subdir/subsubdir/1_1.txt
Scanning Shared Node /tuser/files/groupShareTest/subdir/subsubdir/2_0.txt
Scanning Shared Node /tuser/files/groupShareTest/subdir/subsubdir/1_0.txt
Scanning Shared Node /tuser/files/groupShareTest/subdir/subsubdir/3_1.txt
Scanning Shared Node /tuser/files/groupShareTest/subdir/subsubdir/2_2.txt
Scanning Shared Node /tuser/files/groupShareTest/subdir/subsubdir/3_0.txt
Scanning Shared Node /tuser/files/groupShareTest/subdir/subsubdir/2_1.txt
Finished searching files
Duplicates for user "admin" are: 
1121cfccd5913f0a63fec40a6ffd44ea64f9dc135c66634ba001d10bcf4302a2(file_hash)
     /admin/files/groupShareTest/3_0.txt
     /admin/files/groupShareTest/3_1.txt
     /admin/files/groupShareTest/3_2.txt
     /admin/files/groupShareTest/3_3.txt
     /admin/files/groupShareTest/subdir/3_0.txt
     /admin/files/groupShareTest/subdir/3_1.txt
     /admin/files/groupShareTest/subdir/3_2.txt
     /admin/files/groupShareTest/subdir/3_3.txt
     /admin/files/groupShareTest/subdir/subsubdir/3_0.txt
     /admin/files/groupShareTest/subdir/subsubdir/3_1.txt
     /admin/files/groupShareTest/subdir/subsubdir/3_2.txt
     /admin/files/groupShareTest/subdir/subsubdir/3_3.txt
     /admin/files/tests/3_0.txt
     /admin/files/tests/3_1.txt
     /admin/files/tests/3_2.txt
     /admin/files/tests/3_3.txt
     /admin/files/userShareTest/3_0.txt
     /admin/files/userShareTest/3_1.txt
     /admin/files/userShareTest/3_2.txt
     /admin/files/userShareTest/3_3.txt
     /admin/files/userShareTest/subdir/3_0.txt
     /admin/files/userShareTest/subdir/3_1.txt
     /admin/files/userShareTest/subdir/3_2.txt
     /admin/files/userShareTest/subdir/3_3.txt
     /admin/files/userShareTest/subdir/subsubdir/3_0.txt
     /admin/files/userShareTest/subdir/subsubdir/3_1.txt
     /admin/files/userShareTest/subdir/subsubdir/3_2.txt
     /admin/files/userShareTest/subdir/subsubdir/3_3.txt
4355a46b19d348dc2f57c046f8ef63d4538ebb936000f3c9ee954a27460dd865(file_hash)
     /admin/files/groupShareTest/1_0.txt
     /admin/files/groupShareTest/1_1.txt
     /admin/files/groupShareTest/subdir/1_0.txt
     /admin/files/groupShareTest/subdir/1_1.txt
     /admin/files/groupShareTest/subdir/subsubdir/1_0.txt
     /admin/files/groupShareTest/subdir/subsubdir/1_1.txt
     /admin/files/tests/1_0.txt
     /admin/files/tests/1_1.txt
     /admin/files/userShareTest/1_0.txt
     /admin/files/userShareTest/1_1.txt
     /admin/files/userShareTest/subdir/1_0.txt
     /admin/files/userShareTest/subdir/1_1.txt
     /admin/files/userShareTest/subdir/subsubdir/1_0.txt
     /admin/files/userShareTest/subdir/subsubdir/1_1.txt
53c234e5e8472b6ac51c1ae1cab3fe06fad053beb8ebfd8977b010655bfdd3c3(file_hash)
     /admin/files/groupShareTest/2_0.txt
     /admin/files/groupShareTest/2_1.txt
     /admin/files/groupShareTest/2_2.txt
     /admin/files/groupShareTest/subdir/2_0.txt
     /admin/files/groupShareTest/subdir/2_1.txt
     /admin/files/groupShareTest/subdir/2_2.txt
     /admin/files/groupShareTest/subdir/subsubdir/2_0.txt
     /admin/files/groupShareTest/subdir/subsubdir/2_1.txt
     /admin/files/groupShareTest/subdir/subsubdir/2_2.txt
     /admin/files/tests/2_0.txt
     /admin/files/tests/2_1.txt
     /admin/files/tests/2_2.txt
     /admin/files/userShareTest/2_0.txt
     /admin/files/userShareTest/2_1.txt
     /admin/files/userShareTest/2_2.txt
     /admin/files/userShareTest/subdir/2_0.txt
     /admin/files/userShareTest/subdir/2_1.txt
     /admin/files/userShareTest/subdir/2_2.txt
     /admin/files/userShareTest/subdir/subsubdir/2_0.txt
     /admin/files/userShareTest/subdir/subsubdir/2_1.txt
     /admin/files/userShareTest/subdir/subsubdir/2_2.txt
9a271f2a916b0b6ee6cecb2426f0b3206ef074578be55d9bc94f6f3fe3ab86aa(file_hash)
     /admin/files/groupShareTest/0_0.txt
     /admin/files/groupShareTest/subdir/0_0.txt
     /admin/files/groupShareTest/subdir/subsubdir/0_0.txt
     /admin/files/tests/0_0.txt
     /admin/files/userShareTest/0_0.txt
     /admin/files/userShareTest/subdir/0_0.txt
     /admin/files/userShareTest/subdir/subsubdir/0_0.txt
